components/common/src/util/path.rs

```diff
@@ -92,58 +87,47 @@ async fn interpreter_paths() -> Result<Vec<PathBuf>> {
-                                              {}, yet the parent directory could not be \
-                                              computed. Aborting...",
-                                             INTERPRETER_COMMAND, &path);
-                                    return Err(Error::FileNotFound(path));
-                                }
-                            }
-                        }
-                        None => {
-                            install::type_erased_start(&mut ui::NullUi::new(),
-                                                       &default_bldr_url(),
-                                                       &ChannelIdent::stable(),
-                                                       &(ident.clone(),
-                                                         PackageTarget::active_target())
-                                                                                        .into(),
-                                                       &*PROGRAM_NAME,
-                                                       VERSION,
-                                                       FS_ROOT_PATH.as_path(),
-                                                       &cache_artifact_path(None::<String>),
-                                                       None,
-                                                       &InstallMode::default(),
-                                                       &LocalPackageUsage::default(),
-                                                       InstallHookMode::default()).await?;
-                            let pkg_install =
-                                PackageInstall::load(&ident, Some(FS_ROOT_PATH.as_ref()))?;
-                            pkg_install.paths()?
-                        }
-                    }
+                    if install::type_erased_start(&mut ui::NullUi::new(),
+                                                &default_bldr_url(),
+                                                &ChannelIdent::stable(),
+                                                &(ident.clone(),
+                                                    PackageTarget::active_target())
+                                                                                .into(),
+                                                &*PROGRAM_NAME,
+                                                VERSION,
+                                                FS_ROOT_PATH.as_path(),
+                                                &cache_artifact_path(None::<String>),
+                                                None,
+                                                &InstallMode::default(),
+                                                &LocalPackageUsage::default(),
+                                                InstallHookMode::default()).await.is_err() {
+                                                    return Err(Error::InterpreterNotFound(ident));
```

Comment:
We end up losing the underlying reason _why_ the interpreter could not be found. Maybe this is not a big deal. I have never seen it done before but I wonder if we should do something like `InterpreterNotFound(PackageIdent, Box<Self>)`?

Reply:
good call!